scipy/_lib/_array_api.py

```diff
@@ -0,0 +1,239 @@
+"""Utility functions to use Python Array API compatible libraries.
+
+For the context about the Array API see:
+https://data-apis.org/array-api/latest/purpose_and_scope.html
+
+The SciPy use case of the Array API is described on the following page:
+https://data-apis.org/array-api/latest/use_cases.html#use-case-scipy
+"""
+import math
+import os
+
+import numpy as np
+# probably want to vendor it (submodule)
+import array_api_compat
+from array_api_compat import size
+import array_api_compat.numpy
+
+__all__ = ['array_namespace', 'as_xparray', 'as_xparray_namespace', 'isdtype']
+
+
+# SCIPY_ARRAY_API, array_api_dispatch is used by sklearn
+array_api_dispatch = os.environ.get("array_api_dispatch", False)
+SCIPY_ARRAY_API = os.environ.get("SCIPY_ARRAY_API", array_api_dispatch)
+SCIPY_DEVICE = os.environ.get("SCIPY_DEVICE", "cpu")
+
+_GLOBAL_CONFIG = {
+    "SCIPY_ARRAY_API": SCIPY_ARRAY_API,
+    "SCIPY_DEVICE": SCIPY_DEVICE,
+}
+
+
+def compliance_scipy(*arrays):
```

Comment:
Nit: From the function name it's not immediately clear what this function is doing. Don't have an immediate suggestions though.

Reply:
I also don't have a better suggestion. This is a private utility function only used in this file, so I'll resolve the comment - it doesn't matter much and can be tweaked later if desired.